sdk/identity/azure-identity/README.md

```diff
@@ -26,99 +24,102 @@ pip install azure-identity
+## `DefaultAzureCredential`
+[`DefaultAzureCredential`][default_cred_ref] is appropriate for most
+applications intended to run in Azure. It can authenticate as a service
+principal, managed identity, or user, and  can be configured for local
+development and production environments without code changes.
 
-To authenticate as a service principal, provide configuration in environment
-variables as described in the next section.
+To authenticate as a service principal, provide configuration in
+[environment variables](#environment-variables) as described in the next section.
 
-Authenticating as a managed identity requires no configuration, but does
-require platform support. See the
+Authenticating as a managed identity requires no configuration but is only
+possible on a supported platform. See Azure Active Directory's
 [managed identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/services-support-managed-identities)
 for more information.
 
 ### Single sign-on
-During local development on Windows, `DefaultAzureCredential` can authenticate
-using a single sign-on shared with Microsoft applications, for example Visual
-Studio 2019. Because you may have multiple signed in identities, to
-authenticate this way you must set the environment variable `AZURE_USERNAME`
-with your desired identity's username (typically an email address).
+During local development on Windows, [`DefaultAzureCredential`][default_cred_ref]
+can authenticate using a single sign-on shared with Microsoft applications, for
+example Visual Studio 2019. This may require additional configuration when
+multiple identities have signed in. In that case, set the environment variable
+`AZURE_USERNAME` with the desired identity's username (typically an email
+address).
 
 ## Environment variables
-
-`DefaultAzureCredential` and `EnvironmentCredential` can be configured with
+[`DefaultAzureCredential`][default_cred_ref] and
+[`EnvironmentCredential`][environment_cred_ref] can be configured with
 environment variables. Each type of authentication requires values for specific
 variables:
 
 #### Service principal with secret
 >|variable name|value
 >|-|-
->|`AZURE_CLIENT_ID`|service principal's app id
->|`AZURE_TENANT_ID`|id of the principal's Azure Active Directory tenant
->|`AZURE_CLIENT_SECRET`|one of the service principal's client secrets
+>|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
+>|`AZURE_TENANT_ID`|id of the application's Azure Active Directory tenant
+>|`AZURE_CLIENT_SECRET`|one of the application's client secrets
 
 #### Service principal with certificate
 >|variable name|value
 >|-|-
->|`AZURE_CLIENT_ID`|service principal's app id
->|`AZURE_TENANT_ID`|id of the principal's Azure Active Directory tenant
->|`AZURE_CLIENT_CERTIFICATE_PATH`|path to a PEM-encoded certificate file including private key (without password)
+>|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
```

Comment:
So there's nuance here, and I'm not sure how we should highlight it here. This technically should be service principal id. In most cases the application id and the SP's id should be the same. But if the app is a multi-tenant app they wouldn't be.

Reply:
I struggle with what to call this. People with working knowledge of AAD are fine with "service principal", which I agree is more correct. However, "application" is used all over the place to refer to the same thing. The portal's and CLI's usage of it led me to make this change, hoping "application" can split the difference, helping new users find the right value without confusing the experienced.

Reply:
Your probably right that here in the high level view we should avoid exposing this nuance as multi-tenant apps are not the common case (and not the case we have in the creation commands above). So I like you're change here, but at some point we probably need deeper level docs that goes into some of the nuance.